common/fs/src/cache/dir_path.rs

```diff
@@ -74,3 +83,111 @@ impl From<DirPathBuf> for PathBuf {
         d.inner
     }
 }
+
+// Recursive directory back off
+fn find_valid_path(
+    path: Option<PathBuf>,
+    postfix: Option<PathBuf>,
+) -> Result<DirPathBuf, DirPathBufError> {
+    match path {
+        Some(p) => {
+            if p.is_dir() {
+                Ok(DirPathBuf { inner: p, postfix })
+            } else {
+                warn!("{:?} is not a directory; moving to parent directory", p);
+                let root_pathbuf = Path::new("/");
+                let mut postfix_pathbuf = PathBuf::new();
+                let mut pop_pathbuf = PathBuf::new();
+                if let Some(path) = postfix {
+                    pop_pathbuf.push(path);
+                }
+                let parent = match level_up(&p) {
+                    Some(p) => {
+                        if p == root_pathbuf {
+                            panic!("configured directory recursed to root!");
```

Comment:
Set to panic if function recurses back to the root directory. This will watch the entire root file system which is most likely not desirable.

Reply:
Changed this so that instead of throwing a panic, it will warn and watch root level for missing dir.